feat: add dedicated Mistral AI provider with latest model variants

ThomsenDrake opened the pull request:
- Added Mistral AI provider configuration with 32 models
- Extracted all Mistral models from OpenRouter config
- Added -latest variants for all main model sizes (large, medium, small)
- Added -latest variants for devstral models (medium, small)
- Updated providers.go to include mistralProvider
- Set API endpoint to https://api.mistral.ai/v1

Generated by Mistral Vibe.

ThomsenDrake committed:
feat: add dedicated Mistral AI provider with latest model variants

- Added Mistral AI provider configuration with 32 models
- Extracted all Mistral models from OpenRouter config
- Added -latest variants for all main model sizes (large, medium, small)
- Added -latest variants for devstral models (medium, small)
- Updated providers.go to include mistralProvider
- Set API endpoint to https://api.mistral.ai/v1
- Uses tILXtdeWUWgbTX3Uwg0Flp4SVLEahdyH environment variable

Generated by Mistral Vibe.
Co-Authored-By: Mistral Vibe <vibe@mistral.ai>

ThomsenDrake committed:
fix: update Mistral model pricing based on official pricing tables

- Updated Frontier Models - Generalist pricing:
  - Mistral Large 3: /bin/sh.5/M input, .5/M output
  - Mistral Medium 3.1: /bin/sh.4/M input, /M output
  - Mistral Small 3.2: /bin/sh.1/M input, /bin/sh.3/M output
  - Ministral 3 14B: /bin/sh.2/M input, /bin/sh.2/M output
  - Ministral 3 8B: /bin/sh.15/M input, /bin/sh.15/M output
  - Ministral 3 3B: /bin/sh.1/M input, /bin/sh.1/M output

- Updated Other Models pricing:
  - Devstral Medium 1.0: /bin/sh.4/M input, /M output
  - Devstral Small 1.1: /bin/sh.1/M input, /bin/sh.3/M output
  - Mistral Medium 3: /bin/sh.4/M input, /M output
  - Mistral Large 2.1: /M input, /M output (already correct)
  - Pixtral Large: /M input, /M output (already correct)
  - Mistral Nemo 12B: /bin/sh.15/M input, /bin/sh.15/M output (already correct)
  - Codestral: /bin/sh.3/M input, /bin/sh.9/M output

- Updated -latest variants to match their corresponding models

Generated by Mistral Vibe.
Co-Authored-By: Mistral Vibe <vibe@mistral.ai>

ThomsenDrake committed:
feat: set Devstral models as default for Mistral provider

- Changed default_large_model_id from mistral-large to devstral-medium-latest
- Changed default_small_model_id from mistral-small to devstral-small-latest
- This makes Devstral models the default choice when using the Mistral provider

Generated by Mistral Vibe.
Co-Authored-By: Mistral Vibe <vibe@mistral.ai>

nikaro commented:
Aren't `Devstral 2` and `Devstral Small 2` missing ?

kujtimiihoxha commented:
@ThomsenDrake thanks for the PR, it looks like this pr needs a rebase and also have you made sure that the changes work as expected in crush?

You can test it locally by running the catwalk server `go run .` Than in another tab `export CATWALK_URL=http://localhost:8080/ `and running crush, select the models you have added and just make sure it all works as expected.

This will help me validate the configuration too :)

Commit:
fix(mistral): clean up model names and add provider type constants

- Remove redundant "Mistral: " prefix from non-flagship model names
- Add TypeMistral to known provider types
- Add InferenceProviderMistral to known inference providers
- Register Mistral in KnownProviders() and KnownProviderTypes()

andreynering commented on internal/providers/configs/mistral.json:
```diff
@@ -0,0 +1,427 @@
+  "name": "Mistral AI",
+  "id": "mistral",
+  "api_key": "$MISTRAL_API_KEY",
+  "api_endpoint": "https://api.mistral.ai/v1",
```
Looks like there is a `/models` endpoint available:

https://docs.mistral.ai/api/endpoint/models#operation-list_models_v1_models_get

In those cases, we prefer to have a generator to fetch and update the list for us.

AIs are usually good do create one taking the others as examples. They are all under `cmd/`.

Djiit replied:
Hey @andreynering, I just took at stab at it here: https://github.com/charmbracelet/catwalk/pull/332
